cmd/meroxa/turbine/ruby/deploy.go

```diff
@@ -0,0 +1,114 @@
+package turbinerb
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+
+	"github.com/meroxa/cli/cmd/meroxa/global"
+	utils "github.com/meroxa/cli/cmd/meroxa/turbine"
+)
+
+func (t *turbineRbCLI) NeedsToBuild(ctx context.Context, appName string) (bool, error) {
+	cmd := exec.Command("turbine_client", "hasfunctions", t.appPath)
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		err = fmt.Errorf(
+			"unable to determine if the Meroxa Application at %s has a Process; %s",
+			t.appPath,
+			string(output))
+		return false, err
+	}
+
+	isNeeded, err := utils.GetTurbineResponseFromOutput(string(output))
+	if err != nil {
+		err := fmt.Errorf(
+			"unable to determine if the Meroxa Application at %s has a Process; %s",
+			t.appPath,
+			string(output))
+		return false, err
+	}
+	return strconv.ParseBool(isNeeded)
+}
+
+func (t *turbineRbCLI) Deploy(ctx context.Context, imageName, appName, gitSha, specVersion, accountUUID string) (string, error) {
+	var (
```

Comment:
you'll need to start up the grpc server here in `platform` mode and then get deployment spec from there